### ISSUE TITLE
[libretro] GL driver not working

### ISSUE THREAD

**Issue by ner00** [FCare/Kronos#884]
I don't know if this issue is exclusive to me, but the latest core versions does not work with `gl` video driver for me despite working with `glcore` and `gl1`. The games load fine but I just get a black screen. If I hit F1 to show the quick menu then I can see below the RetroArch menu transparency the game image normally, but once I come back to the game it goes completely black again. Kronos is the only core to have this issue.

I'll post 2 logs, the first working fine using `glcore` driver and the second with no image when using `gl` driver. Two things are immediately apparent, one is that `gl` driver has several lines logging shader information (I don't have any shader enabled afaik), secondly is the line with `[WARN] [WGL]: wglSwapInterval() failed.`

**'glcore' driver**
```
[INFO] Using content: I:\RetroArch\roms\Sega - Saturn\Daytona USA - Championship Circuit Edition (USA)\Daytona USA - Championship Circuit Edition (USA).chd.
[INFO] arg #0: retroarch
[INFO] arg #1: I:\RetroArch\roms\Sega - Saturn\Daytona USA - Championship Circuit Edition (USA)\Daytona USA - Championship Circuit Edition (USA).chd
[INFO] arg #2: -s
[INFO] arg #3: I:\RetroArch\saves
[INFO] arg #4: -S
[INFO] arg #5: I:\RetroArch\states
[INFO] arg #6: -c
[INFO] arg #7: I:\RetroArch\retroarch.cfg
[INFO] arg #8: -L
[INFO] arg #9: I:\RetroArch\cores\kronos_libretro.dll
[INFO] arg #10: -v
[INFO] Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] Unloading game..
[INFO] Unloading core..
[INFO] Unloading core symbols..
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
[INFO] RetroArch 1.8.6 (Git 25be09acd3)
[INFO] Redirecting save file to "I:\RetroArch\saves\Daytona USA - Championship Circuit Edition (USA).srm".
[INFO] Redirecting save state to "I:\RetroArch\states\Daytona USA - Championship Circuit Edition (USA).state".
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz
[INFO] Capabilities:  MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE4.2 AES AVX
[INFO] Built: May  4 2020
[INFO] Version: 1.8.6
[INFO] Git: 25be09acd3
[INFO] =================================================
[INFO] Loading dynamic libretro core from: "I:\RetroArch\cores\kronos_libretro.dll"
[INFO] [Overrides] no core-specific overrides found at I:\RetroArch\config\Kronos\Kronos.cfg.
[INFO] [Overrides] no content-dir-specific overrides found at I:\RetroArch\config\Kronos\Daytona USA - Championship Circuit Edition (USA).cfg.
[INFO] [Overrides] no game-specific overrides found at I:\RetroArch\config\Kronos\Daytona USA - Championship Circuit Edition (USA).cfg.
[INFO] [Environ]: GET_LANGUAGE: "0".
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_INTL.
[INFO] [Environ]: SET_CONTROLLER_INFO.
[INFO] [Remaps]: remap directory: I:\RetroArch\config\remaps
[INFO] [Remaps]: core-specific remap found at I:\RetroArch\config\remaps\Kronos\Kronos.rmp.
[INFO] Redirecting save file to "I:\RetroArch\saves\Daytona USA - Championship Circuit Edition (USA).srm".
[INFO] Redirecting save state to "I:\RetroArch\states\Daytona USA - Championship Circuit Edition (USA).state".
[INFO] [Environ]: GET_LOG_INTERFACE.
[INFO] [Environ]: GET_PERF_INTERFACE.
[INFO] [Environ]: SYSTEM_DIRECTORY: "I:\RetroArch\system".
[INFO] [Environ]: PERFORMANCE_LEVEL: 16.
[INFO] [Environ]: SET_DISK_CONTROL_EXT_INTERFACE.
[INFO] Content loading skipped. Implementation will load it on its own.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: SET_PIXEL_FORMAT: XRGB8888.
[INFO] [Environ]: SET_HW_RENDER.
[INFO] Requesting core OpenGL context (3.3).
[INFO] [RCHEEVOS]: ext_hash 0b8865d4 ('chd')
[INFO] Skipping SRAM load..
[INFO] [RCHEEVOS]: Load task finished
[ERROR] Failed to set initial disk index:
> Expected [1]
> Detected [1]
[INFO] Version of libretro API: 1
[INFO] Compiled against API: 1
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Cheats]: Load game-specific cheatfile: I:\RetroArch\cheats\Kronos\Daytona USA - Championship Circuit Edition (USA).cht
[INFO] [Audio]: Set audio input rate to: 44100.00 Hz.
[INFO] [Video]: Video @ fullscreen
[INFO] [Video]: Using HW render, OpenGL driver forced.
[INFO] [Video]: Using configured "glcore" driver for GL HW render.
[INFO] [GLCore]: Found GL context: wgl
[INFO] [GLCore]: Detecting screen resolution 1920x1080.
[INFO] [WGL] extensions: WGL_ARB_buffer_region WGL_ARB_create_context WGL_ARB_create_context_no_error WGL_ARB_create_context_profile WGL_ARB_create_context_robustness WGL_ARB_context_flush_control WGL_ARB_extensions_string WGL_ARB_make_current_read WGL_ARB_mult
[INFO] [WGL]: Adaptive VSync supported.
[INFO] [WGL]: wglSwapInterval(1)
[INFO] [GLCore]: Initializing HW render (1024 x 1024).
[INFO] [GLCore]: Max texture size: 16384 px, renderbuffer size: 16384 px.
[INFO] [GLCore]: Vendor: NVIDIA Corporation, Renderer: GeForce GTX 760/PCIe/SSE2.
[INFO] [GLCore]: Version: 4.6.0 NVIDIA 445.87.
[INFO] [GLCore]: Using resolution 1920x1080
[INFO] [DINPUT]: Enumerating joypads ...
[INFO] [DINPUT]: Device #0 PID: {E501} VID:{0810}
[WARN] [DINPUT]: Strong rumble unavailable.
[WARN] [DINPUT]: Weak rumble unavailable.
[INFO] [DINPUT]: Done enumerating joypads ...
[INFO] [Joypad]: Found joypad driver: "dinput".
[INFO] [GLCore]: Loading stock shader.
[INFO] [slang]: Building pass #0 (N/A)
[INFO] [GLCore]: Not using frame history.
[INFO] [GLCore]: Not using framebuffer feedback.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Video]: Found display server: win32
[libretro INFO] Using OpenGL 4.6
[libretro INFO] Saturn Emulation mode
[INFO] [XAudio2]: Requesting 64 ms latency, using 64 ms latency.
[INFO] [Menu]: Found menu display driver: "glcore".
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [LED]: LED driver = 'null' 0000000000B98B00
[INFO] [MIDI]: Initializing ...
[INFO] [MIDI]: Input disabled.
[INFO] [MIDI]: Output disabled.
[INFO] [MIDI]: Initialized "winmm" driver.
[INFO] Loading history file: [I:\RetroArch\content_history.lpl].
[INFO] Loading history file: [I:\RetroArch\content_music_history.lpl].
[INFO] Loading history file: [I:\RetroArch\content_video_history.lpl].
[INFO] Loading history file: [I:\RetroArch\content_image_history.lpl].
[INFO] Loading favorites file: [I:\RetroArch\content_favorites.lpl].
[INFO] [GLCore]: VSync => on
[INFO] [WGL]: wglSwapInterval(1)
[INFO] Written to playlist file: I:\RetroArch\content_history.lpl
[INFO] [Environ]: SET_INPUT_DESCRIPTORS:
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] SET_GEOMETRY: 320x224, aspect: 1.333.
[INFO] Paused.
```

**'gl' driver**
```
[INFO] Using content: I:\RetroArch\roms\Sega - Saturn\Daytona USA - Championship Circuit Edition (USA)\Daytona USA - Championship Circuit Edition (USA).chd.
[INFO] arg #0: retroarch
[INFO] arg #1: I:\RetroArch\roms\Sega - Saturn\Daytona USA - Championship Circuit Edition (USA)\Daytona USA - Championship Circuit Edition (USA).chd
[INFO] arg #2: -s
[INFO] arg #3: I:\RetroArch\saves
[INFO] arg #4: -S
[INFO] arg #5: I:\RetroArch\states
[INFO] arg #6: -c
[INFO] arg #7: I:\RetroArch\retroarch.cfg
[INFO] arg #8: -L
[INFO] arg #9: I:\RetroArch\cores\kronos_libretro.dll
[INFO] arg #10: -v
[INFO] Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] Unloading game..
[INFO] Unloading core..
[INFO] Unloading core symbols..
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
[INFO] RetroArch 1.8.6 (Git 25be09acd3)
[INFO] Redirecting save file to "I:\RetroArch\saves\Daytona USA - Championship Circuit Edition (USA).srm".
[INFO] Redirecting save state to "I:\RetroArch\states\Daytona USA - Championship Circuit Edition (USA).state".
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz
[INFO] Capabilities:  MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE4.2 AES AVX
[INFO] Built: May  4 2020
[INFO] Version: 1.8.6
[INFO] Git: 25be09acd3
[INFO] =================================================
[INFO] Loading dynamic libretro core from: "I:\RetroArch\cores\kronos_libretro.dll"
[INFO] [Overrides] no core-specific overrides found at I:\RetroArch\config\Kronos\Kronos.cfg.
[INFO] [Overrides] no content-dir-specific overrides found at I:\RetroArch\config\Kronos\Daytona USA - Championship Circuit Edition (USA).cfg.
[INFO] [Overrides] no game-specific overrides found at I:\RetroArch\config\Kronos\Daytona USA - Championship Circuit Edition (USA).cfg.
[INFO] [Environ]: GET_LANGUAGE: "0".
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_INTL.
[INFO] [Environ]: SET_CONTROLLER_INFO.
[INFO] [Remaps]: remap directory: I:\RetroArch\config\remaps
[INFO] [Remaps]: core-specific remap found at I:\RetroArch\config\remaps\Kronos\Kronos.rmp.
[INFO] Redirecting save file to "I:\RetroArch\saves\Daytona USA - Championship Circuit Edition (USA).srm".
[INFO] Redirecting save state to "I:\RetroArch\states\Daytona USA - Championship Circuit Edition (USA).state".
[INFO] [Environ]: GET_LOG_INTERFACE.
[INFO] [Environ]: GET_PERF_INTERFACE.
[INFO] [Environ]: SYSTEM_DIRECTORY: "I:\RetroArch\system".
[INFO] [Environ]: PERFORMANCE_LEVEL: 16.
[INFO] [Environ]: SET_DISK_CONTROL_EXT_INTERFACE.
[INFO] Content loading skipped. Implementation will load it on its own.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: SET_PIXEL_FORMAT: XRGB8888.
[INFO] [Environ]: SET_HW_RENDER.
[INFO] Requesting core OpenGL context (3.3).
[INFO] [RCHEEVOS]: ext_hash 0b8865d4 ('chd')
[INFO] Skipping SRAM load..
[INFO] [RCHEEVOS]: Load task finished
[ERROR] Failed to set initial disk index:
> Expected [1]
> Detected [1]
[INFO] Version of libretro API: 1
[INFO] Compiled against API: 1
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Cheats]: Load game-specific cheatfile: I:\RetroArch\cheats\Kronos\Daytona USA - Championship Circuit Edition (USA).cht
[INFO] [Audio]: Set audio input rate to: 44100.00 Hz.
[INFO] [Video]: Video @ fullscreen
[INFO] [Video]: Using HW render, OpenGL driver forced.
[INFO] [Video]: Using configured "gl" driver for GL HW render.
[INFO] [GL]: Found GL context: wgl
[INFO] [GL]: Detecting screen resolution 1920x1080.
[INFO] [WGL] extensions: WGL_ARB_buffer_region WGL_ARB_create_context WGL_ARB_create_context_no_error WGL_ARB_create_context_profile WGL_ARB_create_context_robustness WGL_ARB_context_flush_control WGL_ARB_extensions_string WGL_ARB_make_current_read WGL_ARB_mult
[INFO] [WGL]: Adaptive VSync supported.
[INFO] [WGL]: wglSwapInterval(1)
[INFO] [GL]: Vendor: NVIDIA Corporation, Renderer: GeForce GTX 760/PCIe/SSE2.
[INFO] [GL]: Version: 4.6.0 NVIDIA 445.87.
[INFO] [GL]: Using Core GL context, setting up VAO...
[INFO] [GL]: Using resolution 1920x1080
[INFO] [GL]: Default shader backend found: glsl.
[INFO] [Shader driver]: Using GLSL shader backend.
[INFO] [GLSL]: Checking GLSL shader support ...
[WARN] [GL]: Stock GLSL shaders will be used.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] Setting up menu pipeline shaders for XMB ...
[INFO] [GLSL]: Compiling ribbon shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling simple ribbon shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling snow shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling modern snow shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling bokeh shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling snowflake shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Using GLSL version 330.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GL]: Using 1 textures.
[INFO] [GL]: Loaded 1 program(s).
[INFO] [GL]: Initializing HW render (1024 x 1024).
[INFO] [GL]: Max texture size: 16384 px, renderbuffer size: 16384 px.
[INFO] [GL]: Supports FBO (render-to-texture).
[INFO] [DINPUT]: Enumerating joypads ...
[INFO] [DINPUT]: Device #0 PID: {E501} VID:{0810}
[WARN] [DINPUT]: Strong rumble unavailable.
[WARN] [DINPUT]: Weak rumble unavailable.
[INFO] [DINPUT]: Done enumerating joypads ...
[INFO] [Joypad]: Found joypad driver: "dinput".
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Video]: Found display server: win32
[libretro INFO] Using OpenGL 4.6
[libretro INFO] Saturn Emulation mode
[INFO] [XAudio2]: Requesting 64 ms latency, using 64 ms latency.
[INFO] [Menu]: Found menu display driver: "gl".
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [LED]: LED driver = 'null' 0000000000B98B00
[INFO] [MIDI]: Initializing ...
[INFO] [MIDI]: Input disabled.
[INFO] [MIDI]: Output disabled.
[INFO] [MIDI]: Initialized "winmm" driver.
[INFO] Loading history file: [I:\RetroArch\content_history.lpl].
[INFO] Loading history file: [I:\RetroArch\content_music_history.lpl].
[INFO] Loading history file: [I:\RetroArch\content_video_history.lpl].
[INFO] Loading history file: [I:\RetroArch\content_image_history.lpl].
[INFO] Loading favorites file: [I:\RetroArch\content_favorites.lpl].
[INFO] [GL]: VSync => on
[INFO] [WGL]: wglSwapInterval(1)
[WARN] [WGL]: wglSwapInterval() failed.
[INFO] [Environ]: SET_INPUT_DESCRIPTORS:
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] SET_GEOMETRY: 320x224, aspect: 1.333.
[INFO] [GL]: VSync => on
[INFO] [WGL]: wglSwapInterval(1)
```

**Comment by ner00** [comment]
Updated from RetroArch 1.8.6 (Git 25be09acd3) to RetroArch 1.8.6 (Git d3a55ef732) and it now switches (temporarily) automatically to `glcore` driver for Kronos emulation. I assume this is an error detection in RetroArch, not sure what changed and why Kronos is failing to use plain `gl` driver.

**Comment by barbudreadmon** [comment]
@ner00 Yeah, i fixed RA 2 weeks ago, see https://github.com/libretro/RetroArch/issues/10486
Kronos is written for glcore, and as such should get the glcore context it's requesting, the fact it was previously working properly with a gl context and somehow stopped recently is a mystery, but anyway RA's behavior was wrong in the first place.

**Comment by ner00** [comment]
> the fact it was previously working properly with a gl context and somehow stopped recently is a mystery

True. During tests I rolled back to RA v1.8.4 which I remember not having this behaviour with Kronos but with the new Kronos core it always got (non-working) gl all the same. Unfortunately I did not have a backup of the Kronos core to test with, but since glcore was already the intended driver just not being acknowledged by RA, I guess there's no point in going down that rabbit hole.
